bluetooth/src/commonMain/kotlin/device/DeviceConnectionManager.kt

```diff
@@ -18,32 +18,35 @@
 package com.splendo.kaluga.bluetooth.device
 
 import co.touchlab.stately.collections.sharedMutableMapOf
-import co.touchlab.stately.concurrency.AtomicInt
 import co.touchlab.stately.concurrency.AtomicReference
 import co.touchlab.stately.concurrency.value
 import com.splendo.kaluga.bluetooth.Characteristic
 import com.splendo.kaluga.bluetooth.Descriptor
 import com.splendo.kaluga.bluetooth.Service
 import com.splendo.kaluga.bluetooth.UUID
 import com.splendo.kaluga.bluetooth.uuidString
-import com.splendo.kaluga.logging.debug
 import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.channels.BufferOverflow
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.MutableSharedFlow
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.launch
 
 abstract class BaseDeviceConnectionManager(
-    val connectionSettings: ConnectionSettings = ConnectionSettings(),
     val deviceWrapper: DeviceWrapper,
-    val stateRepo: DeviceStateFlowRepo
-) : CoroutineScope by stateRepo {
+    private val bufferCapacity: Int = BUFFER_CAPACITY,
```

Comment:
What this buffer for?

Reply:
The SharedFlow receiving stuff from the device might move faster than the state machine can transition. Using a buffered sharedFlow allows us to not miss data (up to the buffer capacity)